astroplan/core.py

```diff
@@ -0,0 +1,490 @@
+#from math import sqrt, pi, exp, log, floor
+from abc import ABCMeta, abstractmethod
+
+#import numpy as np
+
+# Originally authored by Jazmin Berlanga Medina (jazmin.berlanga@gmail.com),
+# and modified by Adrian Price-Whelan (email) and Erik Tollerud (email).
+
+__all__ = ["Observer", "Target", "FixedTarget", "NonFixedTarget",
+           "Constraint", "TimeWindow", "AltitudeRange",
+           "AboveAirmass", "Observation"]
+
+#__doctest_requires__ = {'*': ['scipy.integrate']}
+
+
+class Observer(object):
+    """
+    Some comments.
+    """
+
+    def __init__(self, name, longitude, latitude, elevation, pressure,
+                 relative_humidity, temperature, timezone, description=None):
+        """
+        Initializes an Observer object.
+
+        <longer description>
+
+        Parameters
+        ----------
+        name : str
+            A short name for the telescope, observatory or location.
+
+        longitude : str or `~astropy.units.quantity`?
+            The longitude of the observing location.
+
+        latitude : str or `~astropy.units.quantity`?
+            The latitude of the observing location.
+
+        elevation : `~astropy.units.Quantity`
+            The elevation of the observing location, with respect to sea
+            level.
+
+        pressure : `~astropy.units.Quantity`
+            The ambient pressure.
+
+        relative_humidity : float
+            The ambient relative humidity.
+
+        temperature : `~astropy.units.Quantity`
+            The ambient temperature.
+
+        timezone : WHAT TYPE IS pytz.timezone ?
+            The local timezone.
+
+        description : str
+            A short description of the telescope, observatory or observing
+            location.
+        """
+        raise NotImplementedError()
+
+    def set_environment(self, pressure, relative_humidity, temperature):
+        """
+        Updates the Observer object with environmental conditions.
+
+        <longer description>
+
+        Parameters
+        ----------
+        pressure : `~astropy.units.Quantity`
+            The ambient pressure.
+
+        relative_humidity : `~astropy.units.Quantity`
+            The ambient relative humidity.
+
+        temperature :
+            The ambient temperature.
+        """
+        raise NotImplementedError()
+
+    def get_date(self, input_date_time, timezone=None):
+        """
+        Builds an object containing date and time information.
+
+        Default time zone is UTC.
+        If time zone (local or otherwise) requested, then uses `datetime.datetime`
+        and pytz.timezone to convert from UTC.
+
+        Must first create date_time object with get_date() before using any other 
+        Observer methods, all of which require one as an argument.
+
+        Parameters
+        ----------
+        input_date_time : WHAT TYPE IS astropy.time OBJECT?
+
+        """
+        raise NotImplementedError()
+
+    # Sun-related methods.
+
+    def noon(date_time):
```

Comment:
For these, you can add a `Returns` block to the docstrings that specifies what you get back -- but, what is returned here? A `datetime` object? @astroplanners/astroplanners-contributors